presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -111,6 +111,21 @@ public boolean matches(Map<ColumnDescriptor, ParquetDictionaryDescriptor> dictio
         return effectivePredicate.overlaps(stripeDomain);
     }
 
+    private Type getType(RichColumnDescriptor column)
+    {
+        // we look at effective predicate domain because it more accurately matches the hive column
+        // than the type available in the parquet metadata passed here as RichColumnDescriptor
+        // for example varchar(len) hive column is translated to binary and then to varchar type using parquet metadata
```

Comment:
> for example varchar(len) hive column is translated to binary.

Can you elaborate please?

Reply:
I took a stab at breaking it down to as much detail as I can. 
1) During predicate match we need the domain type of both the column in the predicate (if specified) and column from schema  to match. [TupleDomainParquetPredicate#matches](https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/TupleDomainParquetPredicate.java#L94)
2) Now for varchar(fixedLen)`effectivePredicate`'s domain  aligns with the type specified in the hive schema. 
3) However the [hive schema columns](https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/TupleDomainParquetPredicate.java#L62) uses RichColumnDescriptor which extends ColumnDescriptor from hive. This class cannot capture the varchar(fixLen) the same way and it doesn't seem straightforward to change that.

Looking at my comment above I can see that I made the mistake of saying ` from parquet metadata`. Let me know if the explanation above make sense and if you have any suggestion to reword that comment. 
Again thanks for offering feedback on this. I think this might be affecting a lot more users than just us.